var/spack/repos/builtin/packages/mummer/package.py

```diff
@@ -39,6 +39,19 @@ class Mummer(Package):
     patch('Makefile.patch')
     patch('scripts-Makefile.patch')
 
+    def patch(self):
+        """Fix mummerplot's use of defined on hashes (deprecated
+           since perl@5.6 made illegal in perl@5.20."""
+
+        # only fix this for versions where it is actually
+        # an error, not just deprecated.
```

Comment:
Waffling, 49/51%.  On the one hand, don't make any changes unless you have to.  On the other hand we might as well fix it.  If I did enough research I think that I can convince myself that I should just *always* change it (I'm pretty sure that the patched form is safe all of the way back.  I'm balancing providing something that works against trying to fix all of the Internet's problems.

Phew...

Your patch for mozjs is independent of the perl version involved.  If that's working out for you then I'd be happy to play along.